### PR TITLE
Add Swift wrappers for some endpoints

### PR DESCRIPTION
**Pull request by jguz-pubnub**

### CODE REVIEW

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/AddChannelsToPush.ios.kt [hunk 1]
```diff
@@ -1,9 +1,41 @@
...
+actual interface AddChannelsToPush : Endpoint<PNPushAddChannelResult>
+
+@OptIn(ExperimentalForeignApi::class)
+open class AddChannelsToPushImpl(
+    private val pubnub: PubNubObjC,
+    private val channels: List<String>,
+    private val deviceId: String
+): AddChannelsToPush {
+    override fun async(callback: Consumer<Result<PNPushAddChannelResult>>) {
+        val deviceIdData = (deviceId as NSString).dataUsingEncoding(NSUTF8StringEncoding)
```
Comment (wkal-pubnub) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/AddChannelsToPush.ios.kt:28]: do you know if this works?
the documentation says it should be possible to cast String to NSString and vice versa, but the IDE complains that this cast doesn't succeed

it's OK if you don't know yet, we can leave it and when we write tests we'll get a crash an fix it

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/PubNubImpl.kt [hunk 2]
```diff
@@ -92,33 +103,59 @@ class PubNubImpl(override val configuration: PNConfiguration) : PubNub {
...
+    @OptIn(ExperimentalForeignApi::class)
     override fun signal(channel: String, message: Any): Signal {
-        TODO("Not yet implemented")
+        return SignalImpl(pubnub = pubNubObjC, channel = channel, message = message)
     }
 
+    @OptIn(ExperimentalForeignApi::class)
     override fun getSubscribedChannels(): List<String> {
-        TODO("Not yet implemented")
+        return pubNubObjC.subscribedChannels() as List<String>
```
Comment (wkal-pubnub) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/PubNubImpl.kt:136]: also did you happen to check if this cast succeeds? (it probably does, but just wanted to know if you ran this code)

**Review by wkal-pubnub — APPROVED** [review]

**Review by marcin-cebo — APPROVED** [review]

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/AddChannelsToPush.ios.kt (hunk 1 above)
Comment (jguz-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/AddChannelsToPush.ios.kt:28]: Sure, I'll verify it and let you know

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/PubNubImpl.kt (hunk 2 above)
Comment (jguz-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/PubNubImpl.kt:136]: FYI, I checked this casting and it works as expected

**Comment by jguz-pubnub** [comment]
Changes are already present in `kmp-dev`

**Pull request closed without merging**